docs/decisions/0053-dotnet-structured-outputs.md

```diff
@@ -0,0 +1,298 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: proposed
+contact: dmytrostruk
+date: 2024-09-10
+deciders: sergeymenshykh, markwallace, rbarreto, westey-m, dmytrostruk, ben.thomas, evan.mattson, crickman
+---
+
+# Structured Outputs implementation in .NET version of Semantic Kernel
+
+## Context and Problem Statement
+
+[Structured Outputs](https://platform.openai.com/docs/guides/structured-outputs) is a feature in OpenAI API that ensures the model will always generate responses based on provided JSON Schema. This gives more control over model responses, allows to avoid model hallucinations and write simpler prompts without a need to be specific about response format. This ADR describes several options how to enable this functionality in .NET version of Semantic Kernel.
```

Comment:
The technology provided by OpenAI Structured Outputs is conceptually the same as that provided by [Guidance](https://github.com/guidance-ai/guidance) which is supported by many open-source models as well as Azure MaS. It might be worth it to consider how this could be applied to these other models / services as well.